controllers/provisioning_controller.go

```diff
@@ -299,16 +299,56 @@ func (r *ProvisioningReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return result, nil
 }
 
-func (r *ProvisioningReconciler) provisioningInfo(provConfig *metal3iov1alpha1.Provisioning, images *provisioning.Images, proxy *osconfigv1.Proxy) *provisioning.ProvisioningInfo {
+func apiServer() (string, error) {
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return "", err
```

Comment:
If an error is encountered at this point, CBO continues and tries again during the next Reconcile(). From my understanding it appears that this condition may not be fixed in a future reconcile.

Reply:
i'll move it to the SetupManager, so if it fails the pod will restart and if it succeeds the host will be cached